src/universal_transfer_operator/integrations/hooks/fivetran.py

```diff
@@ -0,0 +1,365 @@
+        endpoint = self.api_path_destinations + group_id
+        resp = self._do_api_call(("GET", endpoint))
+        return resp["data"]
+
+    def get_groups(self, group_id) -> Any:
+        """
+        Fetches destination information for the given group.
+
+        :param group_id: The Fivetran group ID, returned by a connector API call.
+        """
+        if group_id == "":
+            raise ValueError("No value specified for connector_id")
+        endpoint = self.api_path_groups + group_id
+        resp = self._do_api_call(("GET", endpoint))
+        return resp["data"]
+
+    def check_connector(self, connector_id):
+        """
+        Ensures connector configuration has been completed successfully and is in a functional state.
+
+        :param connector_id: Fivetran connector_id, found in connector settings page in the Fivetran user interface.
+        """
+        connector_details = self.get_connector(connector_id)
+        service_name = connector_details["service"]
+        schema_name = connector_details["schema"]
+        setup_state = connector_details["status"]["setup_state"]
+        if setup_state != "connected":
+            raise AirflowException(
+                f'Fivetran connector "{connector_id}" not correctly configured, '
+                f"status: {setup_state}\nPlease see: "
+                f"{self._connector_ui_url_setup(service_name, schema_name)}"
+            )
+        self.log.info(f"Connector type: {service_name}, connector schema: {schema_name}")
+        self.log.info(f"Connectors logs at {self._connector_ui_url_logs(service_name, schema_name)}")
+        return True
+
+    def set_schedule_type(self, connector_id, schedule_type):
+        """
+        Set connector sync mode to switch sync control between API and UI.
+
+        :param connector_id: Fivetran connector_id, found in connector settings page in the Fivetran user interface.
+        :param schedule_type: Fivetran schedule type
+        """
+        endpoint = self.api_path_connectors + connector_id
+        return self._do_api_call(("PATCH", endpoint), json.dumps({"schedule_type": schedule_type}))
+
+    def prep_connector(self, connector_id, schedule_type):
+        """
+        Prepare the connector to run in Airflow by checking that it exists and is a good state, then update connector
+        sync schedule type if changed.
+
+        :param connector_id: Fivetran connector_id, found in connector settings page in the Fivetran user interface.
+        :param schedule_type: Fivetran schedule type
+        """
+        self.check_connector(connector_id)
+        if schedule_type not in {"manual", "auto"}:
+            raise ValueError('schedule_type must be either "manual" or "auto"')
+        if self.get_connector(connector_id)["schedule_type"] != schedule_type:
+            return self.set_schedule_type(connector_id, schedule_type)
+        return True
+
+    def start_fivetran_sync(self, connector_id):
```

Comment:
Can we add a doc string here on what all the method does?

Reply:
This is the hook code which we will remove once the change is merged on Fivetran providers